### PR TITLE
Issue #912 - Make ResourceNode 'tags' into a more generic 'info' struct

### PR DESCRIPTION
**Pull request by alexmt**
Closes https://github.com/argoproj/argo-cd/issues/912

### CODE REVIEW

File: pkg/apis/application/v1alpha1/generated.proto [hunk 1]
```diff
@@ -391,7 +399,7 @@ message ResourceNode {
 
   optional string name = 5;
 
-  repeated string tags = 6;
+  repeated InfoItem tags = 6;
```
Comment (jessesuen) [inline pkg/apis/application/v1alpha1/generated.proto]: I think we should rename tags. I think application CRD uses info?

Comment (alexmt) [reply pkg/apis/application/v1alpha1/generated.proto]: You are right. Application indeed uses `info`, renaming

**Comment by alexmt** [comment]
Done. @jessesuen PTAL

**Review by gaganapplatix — APPROVED** [review]
LGTM

**Pull request merged**